tfchain_bridge/pkg/bridge/mint.go

```diff
@@ -0,0 +1,160 @@
+package bridge
+
+import (
+	"context"
+	"math/big"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	hProtocol "github.com/stellar/go/protocols/horizon"
+	"github.com/threefoldtech/substrate-client"
+)
+
+// mint handler for stellar
+func (bridge *Bridge) mint(senders map[string]*big.Int, tx hProtocol.Transaction) error {
+	log.Info().Msg("calling mint now")
+
+	if len(senders) > 1 {
+		log.Info().Msgf("cannot process mint transaction, multiple senders found, refunding now")
+		for sender, depositAmount := range senders {
+			return bridge.refund(context.Background(), sender, depositAmount.Int64(), tx)
+		}
+	}
+
+	var receiver string
+	var depositedAmount *big.Int
+	for receiv, amount := range senders {
+		receiver = receiv
+		depositedAmount = amount
+	}
+
+	if tx.Memo == "" {
+		log.Info().Str("tx_id", tx.Hash).Msg("transaction has empty memo, refunding now")
+		return bridge.refund(context.Background(), receiver, depositedAmount.Int64(), tx)
+	}
+
+	if tx.MemoType == "return" {
+		log.Debug().Str("tx_id", tx.Hash).Msg("transaction has a return memo hash, skipping this transaction")
+		// save cursor
+		cursor := tx.PagingToken()
+		err := bridge.blockPersistency.SaveStellarCursor(cursor)
+		if err != nil {
+			log.Err(err).Msg("error while saving cursor")
+			return err
+		}
+		log.Info().Msg("stellar cursor saved")
+		return nil
+	}
+
+	// TODO check if we already minted for this txid
+	minted, err := bridge.subClient.IsMintedAlready(tx.Hash)
+	if err != nil && err != substrate.ErrMintTransactionNotFound {
+		return err
+	}
+
+	if minted {
+		log.Error().Str("tx_id", tx.Hash).Msg("transaction is already minted")
+		return nil
```

Comment:
Either the error log needs to be downgraded to info if this can happen, or if it is unexpected, the nil return needs to be changed to an actual error.